### ISSUE TITLE
Will there be bugs if we use ground truth z depth to supervise the `depth-nerfacto` with depth rank loss?

### ISSUE THREAD

**Issue by hermosaaurora** [nerfstudio-project/nerfstudio#2360]
**Describe the bug**
There may be bugs according to issue #2344, since the `depth` in nerfstudio is indeed distance along the ray. But sometimes depth generated by dense pretrained models or Blender is `z` depth though. If we use it as regularization, I guess there may be some problems.

**Comment by kerrj** [comment]
Hm good question! maybe @alexsax would have some expertise here? I could certainly see this being an issue for depth supervision methods with metric depth.

**Comment by ethanweber** [comment]
So it's taken care of in the metric depth case with the Euclidean check. But yeah, the question is if ordinal relationships are the same when using depth or distance.

**Comment by hermosaaurora** [comment]
Can you give some advice if we want to show z-depth in the browser viewer? I can see the outputs are stored here:

https://github.com/nerfstudio-project/nerfstudio/blob/48135ee4c8e0fb9ac3ab0ea80d2d71042dfb0b41/nerfstudio/models/nerfacto.py#L280-L328

If we want to calculate `z` depth and store it in the `outputs` dict, we need `camera_to_worlds`. But I found `ray_bundle` doesn't have `camera_to_worlds` now. 

It seems that `InputDataset` and `RayGenerator` both have `cameras` attribute:

https://github.com/nerfstudio-project/nerfstudio/blob/48135ee4c8e0fb9ac3ab0ea80d2d71042dfb0b41/nerfstudio/data/datasets/base_dataset.py#L37-L55

https://github.com/nerfstudio-project/nerfstudio/blob/48135ee4c8e0fb9ac3ab0ea80d2d71042dfb0b41/nerfstudio/model_components/ray_generators.py#L26-L41

To figure out how to pass `camera_to_worlds`, I review the training code and found`VanillaPipeline` where `ray_bundle` and `batch` are generated, then the `model_outputs` are generated.

https://github.com/nerfstudio-project/nerfstudio/blob/48135ee4c8e0fb9ac3ab0ea80d2d71042dfb0b41/nerfstudio/pipelines/base_pipeline.py#L290-L315

Can I 
1. store `cameras` in `RayBundle`'s `metadata` attribute, 
2. or just store it in `batch` dict and add an argument `batch` e.g. `def get_outputs(self, ray_bundle, batch)` just as the `Pipeline` base class?

https://github.com/nerfstudio-project/nerfstudio/blob/48135ee4c8e0fb9ac3ab0ea80d2d71042dfb0b41/nerfstudio/pipelines/base_pipeline.py#L137-L151

Maybe the 1st is more natural since if I modify `get_outputs`'s definition, there will be too many changes here. I have no idea if that is expected or not.

**Comment by hermosaaurora** [comment]
> So it's taken care of in the metric depth case with the Euclidean check. But yeah, the question is if ordinal relationships are the same when using depth or distance.

@ethanweber Yeah, I think maybe the ordinal relationships will fail.
